map_gui/src/tools/turn_explorer.rs

```diff
@@ -119,19 +119,17 @@ impl TurnExplorer {
             )
             .draw(ctx),
             Widget::vert_separator(ctx, 50.0),
-            if idx == 0 {
-                Btn::text_fg("<").inactive(ctx)
-            } else {
-                Btn::text_fg("<").build(ctx, "previous turn", Key::LeftArrow)
-            },
+            ctx.style()
+                .btn_prev()
+                .disabled(idx == 0)
+                .build_widget(ctx, "previous turn"),
             Text::from(Line(format!("{}/{}", idx, turns.len())).secondary())
                 .draw(ctx)
                 .centered_vert(),
-            if idx == turns.len() {
-                Btn::text_fg(">").inactive(ctx)
-            } else {
-                Btn::text_fg(">").build(ctx, "next turn", Key::RightArrow)
-            },
+            ctx.style()
+                .btn_next()
+                .disabled(idx == turns.len())
```

Comment:
How do we expose the "Turn Explorer" UI? It's apparently not the same as the Uber-turn explorer.

This seems like a bug - but wanted to leave the old behavior until I could test. (expected `idx + 1 == turns.len()`)

Reply:
Hotkey **z** on a lane (or clicking the lane and using the action button). This tool is also partly used in the `osm_viewer`, although not the GUI panel part.

The correct behavior is actually correct, it's just a bizarre abuse of pagination. 0 means show all the turns, and then the rest are counting the 1st, 2nd, etc turn normally. When looking at an individual turn, you see all the conflicts:
![Screenshot from 2021-01-21 11-00-37](https://user-images.githubusercontent.com/1664407/105399027-e69bd280-5bd7-11eb-9a68-ce8d5378fef6.png)
And the 0th shows the overview:
![Screenshot from 2021-01-21 11-00-54](https://user-images.githubusercontent.com/1664407/105399061-f4e9ee80-5bd7-11eb-989d-75758a689fa7.png)

There's probably a better UI to help dig into the details of individual turns. This is partly a debug UI, although I think it probably has legitimate player use.

Reply:
ah, tricky! Thanks. =)